platform/app/services/syncs/create_service.rb

```
@@ -21,19 +21,29 @@ def fetch_schema
+          source_defined_primary_key: schema[stream]["x-source_defined_primary_key"]
+        )
       end
     end
 
     def determine_sync_mode(stream)
-      if stream["supported_sync_modes"]&.include?("incremental")
-        :incremental_append
+      supported_modes = stream["x-supported_sync_modes"] || []
+      default_mode = stream["x-default_sync_mode"]
+
+      if supported_modes.any? { |mode| mode.to_s.include?("incremental") } && 
```

rubocop (1.69.1)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)